src/test/java/it/pagopa/pn/deliverypush/rest/PnNotificationViewControllerTest.java

```diff
@@ -0,0 +1,93 @@
+package it.pagopa.pn.deliverypush.rest;
+
+import it.pagopa.pn.deliverypush.action.NotificationViewedHandler;
+import it.pagopa.pn.deliverypush.action.it.utils.NotificationTestBuilder;
+import it.pagopa.pn.deliverypush.action.utils.NotificationUtils;
+import it.pagopa.pn.deliverypush.dto.ext.delivery.notification.NotificationInt;
+import it.pagopa.pn.deliverypush.generated.openapi.server.v1.dto.RequestNotificationViewedDto;
+import it.pagopa.pn.deliverypush.generated.openapi.server.v1.dto.ResponseNotificationViewedDto;
+import it.pagopa.pn.deliverypush.service.NotificationService;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import reactor.core.publisher.Mono;
+
+@WebFluxTest(PnNotificationViewController.class)
+public class PnNotificationViewControllerTest {
+
+    private static final String IUN = "fake_iun";
+
+    @MockBean
+    private NotificationService notificationService;
+
+    @MockBean
+    private NotificationUtils notificationUtils;
+
+    @MockBean
+    private NotificationViewedHandler notificationViewedHandler;
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @Test
+    void notifyNotificationViewedNoNotification() {
```

Comment:
Che significa notifyNotificationViewedNoNotification? :-D
Questo è il test che se la notifica viene chiesta dalla radd non viene marcata come letta?

Reply:
Questo testa il caso in cui venga passato uno iun errato che quindi non restituisce nessuna notifica

Reply:
Puoi utilizzare un nome tipo NotExistIUN, FakeIUN....

Reply:
Ok usato FakeIUN